pkg/controllers/user/nsserviceaccount/nssvcaccnt.go

```diff
@@ -0,0 +1,85 @@
+package nsserviceaccount
+
+import (
+	"context"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type namespaceSvcAccountHandler struct {
+	clusterCtx *config.UserContext
+}
+
+func Register(ctx context.Context, cluster *config.UserContext) {
+	logrus.Debugf("Registering namespaceHandler for checking default serviceaccount")
+	nsh := &namespaceSvcAccountHandler{
+		clusterCtx: cluster,
+	}
+	cluster.Core.Namespaces("").AddHandler(ctx, "namespaceHandler", nsh.Sync)
+}
+
+func (nsh *namespaceSvcAccountHandler) Sync(key string, ns *corev1.Namespace) (runtime.Object, error) {
+	if ns == nil || ns.DeletionTimestamp != nil {
+		return nil, nil
+	}
+	logrus.Debugf("namespaceSvcAccountHandler: Sync: key=%v, ns=%+v", key, *ns)
+
+	//handle default svcAccount of system namespaces
+	if err := nsh.handleSystemNS(key); err != nil {
+		logrus.Errorf("namespaceSvcAccountHandler: Sync: error handling default ServiceAccount of namespace key=%v, err=%v", key, err)
+	}
+	return nil, nil
+}
+
+func (nsh *namespaceSvcAccountHandler) handleSystemNS(namespace string) error {
+
+	if namespace == "kube-system" || namespace == "default" || !nsh.isSystemNS(namespace) {
+		return nil
+	}
+
+	var svcAccounts clientv1.ServiceAccountInterface
+	svcAccounts = nsh.clusterCtx.K8sClient.CoreV1().ServiceAccounts(namespace)
+
+	defSvcAccnt, err := svcAccounts.Get("default", metav1.GetOptions{})
+	if err != nil {
+		logrus.Errorf("namespaceSvcAccountHandler: error listing serviceaccount flag: Sync: key=%v, err=%+v", namespace, err)
+		return err
+	}
+
+	if defSvcAccnt.AutomountServiceAccountToken != nil && *defSvcAccnt.AutomountServiceAccountToken == false {
+		return nil
+	}
+
+	automountServiceAccountToken := false
+	defSvcAccnt.AutomountServiceAccountToken = &automountServiceAccountToken
+	defSvcAccnt.Namespace = namespace
+	logrus.Debugf("namespaceSvcAccountHandler: updating default serviceaccount key=%v", defSvcAccnt)
+	_, err = svcAccounts.Update(defSvcAccnt)
+	if err != nil {
+		logrus.Errorf("namespaceSvcAccountHandler: error updating serviceaccnt flag: Sync: key=%v, err=%+v", namespace, err)
+		return err
+	}
+
+	return nil
+}
+
+func (nsh *namespaceSvcAccountHandler) isSystemNS(namespace string) bool {
+	systemNamespacesStr := settings.SystemNamespaces.Get()
```

Comment:
Well, sure. But what does the check get you that just processing the string doesn't? Either way we return false and don't touch the namespace. If someone sets it to blank they have much bigger problems.

Reply:
It would buy you an error as opposed to a false negative. But agree that it's not really a bad outcome and there would be bigger problems than this, hence the nit.